trainers/base_trainer.py

```diff
@@ -33,30 +39,36 @@ def __init__(
         optimizer,
         dataloader: train_dataloader.BaseDataloader,
         loss_fn,
+        gpu_id, ## new
         lr_scheduler=None,
         dropout_scheduler=None,
     ) -> None:
         self.model = model
+        self.model = DDP(self.model, device_ids=[gpu_id]) ## new
```

Comment:
wonder what the effect of having this as a seperate attribute of trainer would be i.e. self.ddp_model = ... -- cos then we can still pass the underlying model to other methods, and just change how we call the forward pass to call self.ddp_model